src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirOperationDataStore.cs

```diff
@@ -97,7 +74,51 @@ public Task<ExportJobOutcome> GetExportJobByHashAsync(string hash, CancellationT
 
         public Task<ExportJobOutcome> UpdateExportJobAsync(ExportJobRecord jobRecord, WeakETag eTag, CancellationToken cancellationToken)
         {
-            throw new NotImplementedException();
+            // TODO: Implement this method.
+            return Task.FromResult(new ExportJobOutcome(jobRecord, eTag));
+        }
+
+        public async Task<IReadOnlyCollection<ExportJobOutcome>> AcquireExportJobsAsync(ushort maximumNumberOfConcurrentJobsAllowed, TimeSpan jobHeartbeatTimeoutThreshold, CancellationToken cancellationToken)
+        {
+            // We will consider a job to be stale if its timestamp is smaller than or equal to this.
+            DateTimeOffset expirationTime = Clock.UtcNow - jobHeartbeatTimeoutThreshold;
+
+            // We will timestamp the jobs when we mark them as running to track stale jobs.
+            DateTimeOffset heartbeatTimeStamp = Clock.UtcNow;
+
+            using (SqlConnectionWrapper sqlConnectionWrapper = _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapper(true))
+            using (SqlCommand sqlCommand = sqlConnectionWrapper.CreateSqlCommand())
+            {
+                V1.AcquireExportJobs.PopulateCommand(
+                    sqlCommand,
+                    expirationTime,
+                    maximumNumberOfConcurrentJobsAllowed,
+                    heartbeatTimeStamp);
+
+                var acquiredJobs = new List<ExportJobOutcome>();
+
+                using (SqlDataReader sqlDataReader = await sqlCommand.ExecuteReaderAsync(CommandBehavior.SequentialAccess, cancellationToken))
+                {
+                    while (await sqlDataReader.ReadAsync(cancellationToken))
+                    {
+                        (string rawJobRecord, byte[] rowVersionAsBytes) = sqlDataReader.ReadRow(V1.ExportJob.RawJobRecord, V1.ExportJob.JobVersion);
+
+                        var exportJobRecord = JsonConvert.DeserializeObject<ExportJobRecord>(rawJobRecord);
+
+                        if (BitConverter.IsLittleEndian)
```

Comment:
You could consider doing conversion in SQL where you don't have to worry about endianess: `CONVERT(bigint, JobVersion)`.

Reply:
I agree that returning `bigint` would be cleaner. I'm able to implement this - however I would no longer be able to use the `SqlDataReader.ReadRow(column0, column1...)` method to read the value (and would need to opt for `SqlDataReader.GetInt64(columnName, ordinal)` instead).

I decided to leave the conversion in the C# code for now but am happy to change it if you think the alternative is better practice.